gdx/src/com/badlogic/gdx/scenes/scene2d/Group.java

```diff
@@ -237,17 +248,42 @@ public Rectangle getCullingArea () {
 		return cullingArea;
 	}
 
+	public HitMode getHitMode () {
+		return hitMode;
+	}
+
+	/** Determines how to choose the child that has been hit in this group. Default is {@link HitMode#first}. */
+	public void setHitMode (HitMode hitMode) {
+		if (hitMode == null) throw new IllegalArgumentException("hitMode cannot be null.");
+		this.hitMode = hitMode;
+	}
+
 	public Actor hit (float x, float y, boolean touchable) {
 		if (touchable && getTouchable() == Touchable.disabled) return null;
 		Vector2 point = tmp;
+		float hitActorDst = Float.MAX_VALUE;
+		Actor hitActor = null;
 		Actor[] childrenArray = children.items;
 		for (int i = children.size - 1; i >= 0; i--) {
 			Actor child = childrenArray[i];
 			if (!child.isVisible()) continue;
 			child.parentToLocalCoordinates(point.set(x, y));
 			Actor hit = child.hit(point.x, point.y, touchable);
-			if (hit != null) return hit;
+			if (hit != null) {
+				switch (hitMode) {
```

Comment:
Should the switch be outside the loop? There could be hundreds of children to check. Although maybe the difference is negligible due to branch prediction. Maybe make a local `final` reference to the setting and use that.